### PR TITLE
Fix incorrect completions when identifier overlaps with keyword

### PR DESCRIPTION
**Pull request by OskarDamkjaer**
We have logic for knowing when to complete the next vs current word. Our logic only moved the cursor back for identifier, so it didn't work when there was overlap between identfiers and keywords. For example:

`SHOW DAT` correctly suggests `DATABASE`
but
`SHOW DATA` would not.

### CODE REVIEW

**Review by ncordon — APPROVED** [review]

File: .changeset/stale-donkeys-grow.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@neo4j-cypher/language-support': patch
+---
+
+Fix incorrect completions when identifier overlaps with keyword
```
Comment (ncordon) [inline .changeset/stale-donkeys-grow.md:5]: Only comment is this should not be included in the pr I think and we should bunch them up before doing a release? Rather than creating this patch.

Comment (OskarDamkjaer) [reply .changeset/stale-donkeys-grow.md:5]: This will not trigger a release, only add a changeset right? It's nice to add the changeset relating to the commit at the same time we do the commit IMO
